drivers/bmx055/bmx055.c

```diff
@@ -0,0 +1,370 @@
+/*
+ * Copyright (C) 2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_bmx055 BMX055 9-axis gyroscope
+ * @ingroup     drivers_sensors
+ * @brief       Device driver for the BMX055 9-axis gyroscope
+ * @{
+ *
+ * @file
+ * @brief       Device driver interface for the BMX055 gyroscope
+ *
+ * @note The current state of the driver only implements a very basic polling.
+ *
+ * @author      Semjon Kerner <semjon.kerner@fu-berlin.de>
+ */
+
+#include <string.h>
+#include <stdint.h>
+#include <math.h>
+
+#include "periph/i2c.h"
+#include "periph/gpio.h"
+#include "assert.h"
+
+#include "bmx055.h"
+#include "bmx055_internal.h"
+
+#define ENABLE_DEBUG    (1)
+#include "debug.h"
+
+#define BUS_CLK  		(I2C_SPEED_FAST)
+#define BUS  			(dev->p.i2c)
+#define ADDR_MAG		(dev->p.addr_mag)
+#define ADDR_ACC		(dev->p.addr_acc)
+#define ADDR_GYRO		(dev->p.addr_gyro)
+#define GYRO_2000_DPS   (2000U)
+#define GYRO_MAX_SCALE  (0x7FFFU)
+
+/**
```

Comment:
The doxygen block is a duplicate from the header one. Same for all functions in this file